testing/unittests/jthreadpooltests.cpp

```diff
@@ -0,0 +1,943 @@
+        // The two threads should start immediately
+        beforeStart = std::chrono::high_resolution_clock::now();
+        params = {lifespan1000milliseconds, "Thread7"};
+        pool->start(&params, params.name, INFINITE);
+        params = {lifespan1000milliseconds, "Thread8"};
+        pool->start(&params, params.name, INFINITE);
+        duration = measureThreadStartDuration(beforeStart, 2);
+        CPPUNIT_ASSERT(duration.count() < 1000);
+        CPPUNIT_ASSERT_EQUAL(7U, threadStartedCount.load());
+        CPPUNIT_ASSERT_EQUAL(2U, pool->runningCount());
+
+        // Thread9 should start after one of the above completes
+        std::thread asyncStart([this, &beforeStart]()
+                               {
+                                   beforeStart = std::chrono::high_resolution_clock::now();
+                                   ThreadParams asyncParams = {lifespan1000milliseconds, "Thread9"};
+                                   this->pool->start(&asyncParams, asyncParams.name, INFINITE); });
+        asyncStart.detach();
+        duration = measureThreadStartDuration(beforeStart, 1);
+        CPPUNIT_ASSERT(duration.count() > 990);
+        CPPUNIT_ASSERT_EQUAL(8U, threadStartedCount.load());
+        CPPUNIT_ASSERT(threadCompletedCount.load() >= 1);
+        CPPUNIT_ASSERT(pool->runningCount() >= 1);
+
+        // Test multiple threads with INFINITE timeout waiting
+        beforeStart = std::chrono::high_resolution_clock::now();
+        std::thread asyncStart4([this]()
+                                {
+                                    ThreadParams asyncParams = {lifespan1000milliseconds, "Thread10"};
+                                    this->pool->start(&asyncParams, asyncParams.name, INFINITE); });
+
+        std::thread asyncStart5([this]()
+                                {
+                                    ThreadParams asyncParams = {lifespan1000milliseconds, "Thread11"};
+                                    this->pool->start(&asyncParams, asyncParams.name, INFINITE); });
+
+        asyncStart4.detach();
+        asyncStart5.detach();
+
+        // Wait for Thread4 and Thread5 to start (they should wait for slots to become available)
+        duration = measureThreadStartDuration(beforeStart, 2);
+        CPPUNIT_ASSERT(duration.count() > 990);
+        CPPUNIT_ASSERT_EQUAL(10U, threadStartedCount.load());
+        CPPUNIT_ASSERT_EQUAL(8U, threadCompletedCount.load());
+
+        // All threads should complete
+        pool->joinAll(true);
+        CPPUNIT_ASSERT_EQUAL(10U, threadStartedCount.load());
+        CPPUNIT_ASSERT_EQUAL(10U, threadCompletedCount.load());
+        CPPUNIT_ASSERT_EQUAL(0U, pool->runningCount());
+        CPPUNIT_ASSERT(pool->running());
+    }
+
+    void testThrottledThreadPoolWithSpecifiedDelay()
+    {
+        // Test: Expected throttling delays are introduced when pool is at capacity
+        pool.setown(createThreadPool("ThrottledThreadPool", factory, true, nullptr, 3, 500));
```

Comment:
stopped reviewing this test beyond this line, but it sounds like the same as previous except with .5 s vs default 1s ?
If so, it is should at least be combined into a single helper, but on the face of it, it doesn't sounds like it's testing anything very different